src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -238,10 +238,18 @@ def get_facts_from_pdb(nodename, environment)
       # Initialize our server fact hash; we add these to each catalog, and they
       # won't change while we're running, so it's safe to cache the values.
       #
-      # This is lifted directly from Puppet::Indirector::Catalog::Compiler.
+      # See also set_server_facts in Puppet::Indirector::Catalog::Compiler in puppet.
       def set_server_facts
         @server_facts = {}
 
+        # Add our server Puppet Enterprise version, if available.
+        pe_version_file = '/opt/puppetlabs/server/pe_version'
```

Comment:
Is this path a setting somewhere we could read?

Reply:
It appears to be hardcoded everywhere:

https://github.com/puppetlabs/puppet-enterprise-modules/search?q=%22%2Fopt%2Fpuppetlabs%2Fserver%2Fpe_version%22&unscoped_q=%22%2Fopt%2Fpuppetlabs%2Fserver%2Fpe_version%22

Reply:
![](https://media.giphy.com/media/Y4z9olnoVl5QI/giphy.gif)

Reply:
It's "safe" because it's hardcoded everywhere :(